demo/demo_primary.py

```diff
@@ -329,19 +329,25 @@ def update_cycle():
   try:
     primary_ecu.primary_update_cycle()
 
-  # Print a REPLAY banner if ReplayedMetadataError is raised by
-  # primary_update_cycle() while performing an update from the director
-  # repository.  This banner is strictly a demo check for replayed metadata for
-  # Timestamp metadata, and all others should be re-raised.
+  # Print a REPLAY or DEFENDED banner if ReplayedMetadataError or
+  # BadSignatureError is raised by primary_update_cycle().  These banners are
+  # only triggered for bad Timestamp metadata, and all other exception are
+  # re-raised.
   except tuf.NoWorkingMirrorError as exception:
     director_file = os.path.join(_vin, 'metadata', 'timestamp' + demo.METADATA_EXTENSION)
     for mirror_url in exception.mirror_errors:
       if mirror_url.endswith(director_file):
         if isinstance(exception.mirror_errors[mirror_url], tuf.ReplayedMetadataError):
           print_banner(BANNER_REPLAY, color=WHITE+BLACK_BG,
               text='The Director has instructed us to download a Timestamp'
-              ' that is older than the currently trusted version. This '
```

Comment:
But we specifically check for the Timestamp file on line 337.  This banner should never display for non-Timestamp metadata